system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -173,13 +189,49 @@ this.TopStoriesFeed = class TopStoriesFeed {
 
   // Need to remove parenthesis from image URLs as React will otherwise
   // fail to render them properly as part of the card template.
-  _normalizeUrl(url) {
+  normalizeUrl(url) {
     if (url) {
       return url.replace(/\(/g, "%28").replace(/\)/g, "%29");
     }
     return url;
   }
 
+  maybeAddSpoc(target) {
+    if (!this.show_spocs) {
+      return;
+    }
+
+    if (this.newTabsSinceSpoc === 0 || this.newTabsSinceSpoc === this.spocsPerNewTabs) {
+      const updateContent = () => {
+        if (!this.spocs || !this.spocs.length) {
+          // We have stories but no spocs so there's nothing to do and this update can be
+          // removed from the queue.
+          return false;
+        }
+
+        // Create a new array with a spoc inserted at index 2
+        // For now we're using the top scored spoc until we can support viewability based rotation
+        let rows = this.stories.slice(0, this.stories.length);
+        rows.splice(2, 0, this.spocs[0]);
+
+        // Send a content update to the target tab
+        const action = {type: at.SECTION_UPDATE, meta: {skipMain: true}, data: Object.assign({rows}, {id: SECTION_ID})};
```

Comment:
@csadilek  Should this action instead be triggered through the `SectionsManager.updateSection`? To ensure all section updates behave in the same way? Unless there's something specific happening here.

Reply:
@piatra The specific thing here is the skipMain: true. This will update the content of single tab directly without affecting the state in the sectionsmanager or reducer. This is by design. We only want to swap out card 3 temporarily. So, I think this is fine for the experiment?

Reply:
👍 ok thanks. I was just looking into all SECTION_UPDATE calls for a different bug.